app/Community/Components/UserProfileMeta.php

```diff
@@ -0,0 +1,227 @@
+        $rankOffset = (int) (($rank - 1) / 25) * 25;
+
+        return compact(
+            'rank',
+            'numRankedUsers',
+            'rankPercent',
+            'rankPercentLabel',
+            'rankOffset',
+        );
+    }
+
+    private function buildDeveloperStats(User $user): array
+    {
+        $gameAuthoredAchievementsCount = $user->authoredAchievements()
+            ->published()
+            ->select(DB::raw('COUNT(DISTINCT GameID) as game_count'))
+            ->first()
+            ->game_count;
+
+        $totalAuthoredLeaderboards = $user->authoredLeaderboards()
+            ->select(DB::raw('COUNT(LeaderboardDef.ID) AS TotalAuthoredLeaderboards'))
+            ->value('TotalAuthoredLeaderboards');
+
+        $totalAuthoredCodeNotes = $user->authoredCodeNotes()->where('Note', '!=', '')->count();
+
+        $openTickets = null;
+        // FIXME: Uses legacy roles.
+        if ($user->getAttribute('Permissions') >= Permissions::Registered) {
+            $openTickets = array_sum(countOpenTicketsByDev($user->User));
+        }
+
+        $ticketsClosedResolved = getNumberOfTicketsClosedForOthers($user->User);
+        $ticketsResolvedForOthers = 0;
+        foreach ($ticketsClosedResolved as $ticketData) {
+            $ticketsResolvedForOthers += $ticketData['ResolvedCount'];
+        }
+
+        return compact(
+            'gameAuthoredAchievementsCount',
+            'openTickets',
+            'ticketsResolvedForOthers',
+            'totalAuthoredCodeNotes',
+            'totalAuthoredLeaderboards',
+        );
+    }
+
+    private function buildSocialStats(User $user): array
+    {
+        $userSetRequestInformation = getUserRequestsInformation($user);
+        $numForumPosts = $user->forumPosts()->count();
+
+        return compact(
+            'numForumPosts',
+            'userSetRequestInformation',
+        );
+    }
+
+    private function calculateAverageFinishedGames(array $userJoinedGamesAndAwards): string
+    {
+        $totalGames = count($userJoinedGamesAndAwards);
+        $finishedGames = 0;
+
+        // Iterate over each game to check if it is finished.
+        foreach ($userJoinedGamesAndAwards as $game) {
+            if (isset($game['HighestAwardKind']) && $game['HighestAwardKind'] !== null) {
+                $finishedGames++;
+            }
+        }
+
+        // Calculate the average percentage of finished games.
+        $averageFinishedGames = 0;
+        if ($totalGames > 0) {
+            $averageFinishedGames = ($finishedGames / $totalGames) * 100;
+        }
+
+        // Format and return the result to 2 decimal places.
+        return number_format($averageFinishedGames, 2, '.', '');
+    }
+
+    private function calculateAveragePointsPerWeek(string $userMemberSince, int $points = 0): int
+    {
+        $memberSince = Carbon::createFromFormat('Y-m-d H:i:s', $userMemberSince);
+        $now = Carbon::now();
+
+        $weeksOfMembership = $memberSince->diffInWeeks($now);
```

Comment:
Should this use `$memberSince->diffInWeeks($user->lastLogin)`? 

If a user hasn't logged in for many years, should their average points per week reflect that? 

If they did come back, it would obviously tank the value.

Reply:
I lean towards no but will defer to what decision you believe is best. My "no" leaning is only because this sounds like something we'd need to explain to people, which probably makes it a difficult UX to reason about.

Reply:
I don't have strong opinions one way or the other. It might be weird to see Average Points per Week decay on a retired user, but would someone even notice?

Reply:
Barring any objections, for now I'll keep this as-is.